src/trace.ml

```diff
@@ -589,31 +620,53 @@ module Make_commands (Backend : Backend_intf.S) = struct
            | Some path -> path
            | None -> failwithf "Can't find executable for %s" prog ()
          in
-         record_opt_fn ~executable ~f:(fun opts ->
-             let elf = Elf.create opts.executable in
-             let%bind range_symbols =
-               evaluate_trace_filter ~trace_filter:opts.trace_filter ~elf
-             in
-             let%bind pid =
-               let argv = prog :: List.concat (Option.to_list argv) in
-               run_and_record
-                 opts
+         let%bind () =
+           record_opt_fn ~executable ~f:(fun opts ->
+               let elf = Elf.create opts.executable in
+               let%bind range_symbols =
+                 evaluate_trace_filter ~trace_filter:opts.trace_filter ~elf
+               in
+               let%bind pid =
+                 let argv = prog :: List.concat (Option.to_list argv) in
+                 run_and_record
+                   opts
+                   ~elf
+                   ~debug_print_perf_commands
+                   ~prog
+                   ~argv
+                   ~collection_mode:opts.collection_mode
+               in
+               let%bind.Deferred perf_maps = Perf_map.Table.load_by_pids [ pid ] in
+               decode_to_trace
+                 ~perf_maps
+                 ?range_symbols
                  ~elf
+                 ~trace_scope:opts.trace_scope
                  ~debug_print_perf_commands
-                 ~prog
-                 ~argv
+                 ~record_dir:opts.record_dir
                  ~collection_mode:opts.collection_mode
-             in
-             let%bind.Deferred perf_maps = Perf_map.Table.load_by_pids [ pid ] in
-             decode_to_trace
-               ~perf_maps
-               ?range_symbols
-               ~elf
-               ~trace_scope:opts.trace_scope
-               ~debug_print_perf_commands
-               ~record_dir:opts.record_dir
-               ~collection_mode:opts.collection_mode
-               decode_opts))
+                 decode_opts)
+         in
+         let%bind.Deferred () =
```

Comment:
Hmm, I don't think we should do this. Oftentimes we'll run `magic-trace attach ... -serve` through SSH, so there'd be no browser to open and this'd fail.

Reply:
Do you recommend catching an error here or something else? I was unable to find a way to detect from inside the function whether there is actually a browser available.

Reply:
I think we just shouldn't be opening the user's browser. We already print a link to the terminal for where the user should go to view their trace, and many terminals just allow clicking into it directly.